model/src/main/proto/platform_parameter.proto

```diff
@@ -6,32 +6,39 @@ option java_package = "org.oppia.android.app.model";
 option java_multiple_files = true;
 
 // It will correspond to a single Platform Parameter Value.
-message Parameter {
+message PlatformParameter {
     // Value of a parameter can be of any one of the following type.
     oneof value {
         // Boolean type parameter value.
-        bool boolean = 3;
+        bool boolean = 2;
         // Integer type parameter value.
-        int32 integer = 2;
+        int32 integer = 3;
         // String type parameter value.
         string string = 4;
     }
 }
 
 // Storage format of platform parameters while the app is running.
-message ParameterMap{
-    map<string,Parameter> parameter = 1;
+message PlatformParameterMap{
+    map<string,PlatformParameter> parameter = 1;
 }
 
 // Pair of Platform Parameter Value and Name.
-message ParameterWithName{
+message PlatformParameterWithName{
```

Comment:
Ah I wonder if naming this `PlatformParameter` and naming the above `PlatformParameter` => `PlatformParamaterValue` or `PlatformParameterValueWrapper` might better represent these data objects. Since this object itself is identifying the parameter flag (both its name and its value) and the above proto defines the value type and abstracts it out for the map. This would look something like:
```
message PlatformParameterValueWrapper { oneof value {...} }

message PlatformParamterMap { map<string, PlatformParameterValueWrapper> }

message PlatformParamter { string name, oneof value {...} }
```

Reply:
@jcqli although I am not sure about the names but we will have to revert back to the original structures. Because while retrieving the values from the ParameterDatabase (in a domain level controller) we will get individual PlatfromParameterWithName type objects (which you are calling in the comment above as PlatformParameter). These objects will be needed to be stored in a PlatformParameterMap before using them. But PlatformParameterMap is map of String and PlatformParameter (which you are calling in the comment above as PlatformParameterValueWrapper). Hence we will be needed to first take out the value from PlatformParameterWithName and then make a PlatformParameter object out of it before getting mapped. This could have been avoided if we went back to previous structure of storing an Object of PlatformParameter in the PlatformParameterWithName Object. @vinitamurthi WDYT?

Reply:
Yup, this was the tradeoff I was thinking about with data type conversions. I think you're right that using the PlatformParameter proto would be better and streamline the conversion, though I think that you'll have to convert to / from these types later on anyways when receiving the JSON and when using the values.

Reply:
Yeah agreed.